tiledb/sm/misc/comparators.h

```diff
@@ -300,7 +300,8 @@ class GlobalCmp : protected CellCmpBase {
       }
     } else {  // COL_MAJOR
       assert(tile_order_ == Layout::COL_MAJOR);
-      for (unsigned d = dim_num_ - 1;; --d) {
+      unsigned d = dim_num_ - 1;
+      while (d-- > 0) {
```

Comment:
❤️

Reply:
Other parts of the code (ie: dense_reader.cc will do for (int32_t d = static_cast<int32_t>(dim_num) - 1; d >= 0; d--)... Possibly change to that for consistency.